### DISCUSSION TITLE
Wait Node in Netflix Conductor

### DISCUSSION THREAD

**Discussion by swapnil-pando** [Netflix/conductor#3050] [General]
Hi regarding Netflix conductor, how to unpause a wait node. Also in the wait node I have to pass some payload to be used in next step. Is it possible to do in conductor?
Let’s say there is a Web application for shipment creation. Once a shipment is created it has its own life cycle as mentioned below:
Shipment Request Creation
In this step, a user fills out a form to create a request in a web application, and from the backend API service, I would like to trigger a workflow that contains Shipment Request Creation node. The trigger will have a JSON payload of a shipment request.
Shipment update ERP
Post the output of the previous node to the ERP system.
For this step, I could use the HTTP Request node.
Step 1 and Step 2 will happen immediately once the workflow is triggered.
Vehicle Assignment
In this step, a user fills vehicle details against a specific shipment request in a web application, and from the backend API service, I would like to trigger a workflow that contains Vehicle Assignment node. The trigger will have a JSON payload of a shipment request.
This step should get executed only when user action happens at other web applications and how to create a custom node which should wait for such trigger.
Truck-In
In this step, a user fills truck details during entry to the factory against a specific shipment request in a web application, and from the backend API service, I would like to trigger a workflow that contains Truck In node. The trigger will have a JSON payload of a shipment request.
This step should get executed only when user action happens at other web applications and how to create a custom node which should wait for such trigger.
Truck Out
In this step, a user fills truck details during exit from the factory against a specific shipment request in a web application, and from the backend API service, I would like to trigger a workflow that contains Truck Out node. The trigger will have a JSON payload of a shipment request.
This step should get executed only when user action happens at other web applications and how to create a custom node which should wait for such trigger.
Shipment Provision to ERP
Post the output of the previous node to the ERP system.
For this step, I could use the HTTP Request node and can execute it immediately after the previous node execution.
Proof of Delivery
In this step, a user fills delivery details against a specific shipment request in a web application, and from the backend API service, I would like to trigger a workflow that contains Proof of Delivery node. The trigger will have a JSON payload of a shipment request.
This step should get executed only when user action happens at other web applications and how to create a custom node which should wait for such trigger.
Vehicle Assignment, Truck-In, Truck Out, Proof of Delivery are needs to be wait nodes in a single workflow and should be able to trigger via HTTP requests from other applications.
This is the scenario I am trying to implement. For user action or click of a button, I need a wait node or a pause node. Is it possible to do so in conductor?

**Comment by manan164** [comment]
Hi @swapnil-pando , Here is the [documentation ](https://conductor.netflix.com/reference-docs/wait-task.html)to update the WAIT tasks. Here is the corresponding [handler ](https://github.com/Netflix/conductor/blob/main/rest/src/main/java/com/netflix/conductor/rest/controllers/QueueAdminResource.java#L72)for that. We can pass output as params in the api call and the same can be used by following tasks as `${wait_task_refe_name.output.truck_in_time}`

**Reply by swapnil-pando** [reply]
Hi @manan164 @dougsillars , thanks for the reply. I had some additional questions related to the flow I want to implement. First one is now I have a http request which creates a shipment number and based on this shipment number, in the next step the url  for another event in the flow has to be dynamically decided based on the shipment number received in the previous step. Now, in some other workflow engines the solution was to save this variable as a context variable, but that was not helpful to me since every trigger of that workflow engine did not initiate a parallel execution and hence  the context variable was getting overwritten. Is there any way to solution this in conductor?

**Reply by manan164** [reply]
Hi @swapnil-pando , Yeah it is possible in the conductor. Based on the shipment number if URL has small and deterministic changes then it can be done using [SWITCH ](https://conductor.netflix.com/reference-docs/switch-task.html) task. If we need pure dynamic urls based on shipment number then we can have a worker task which can consume the shipment number and return the url. This worker task can be implemented in Java, C#, go, clojure and python. Here more [documentation ](https://conductor.netflix.com/how-tos/csharp-sdk.html)related to worker task.

**Comment by dougsillars** [comment]
@manan164 *just* beat me.. but here are a few more details that might help.

The WAIT system task is exactly what you are looking for.
https://conductor.netflix.com/reference-docs/wait-task.html

Here is a sample wait task:

```
 {
      "name": "waiting_around",
      "taskReferenceName": "waiting_around_ref",
      "type": "WAIT",
    }
```

When your workflow is running and it gets to this task - it will sit and wait.  

When the user has entered the data you want, you can  send a POST to:

`api/queue/update/{workflowId}/waiting_around_ref/COMPLETED`

You can append the data you want in the body of the POST {"shipping_request":"12345", "tracking_number":"1Z237632249924042"}

This JSON will appear as the output of the task, and can be used further along in the workflow.
